opentelemetry-api/src/opentelemetry/configuration/__init__.py

```diff
@@ -18,12 +18,42 @@
 """
 Simple configuration manager
 
-This is a configuration manager for the Tracer and Meter providers. It reads
-configuration from environment variables prefixed with
-``OPENTELEMETRY_PYTHON_``:
+This is a configuration manager for OpenTelemetry. It reads configuration
+values from environment variables prefixed with
+``OPENTELEMETRY_PYTHON_`` whose characters are only all caps and underscores.
+The first character after ``OPENTELEMETRY_PYTHON_`` must be an uppercase
+character.
 
-1. ``OPENTELEMETRY_PYTHON_TRACER_PROVIDER``
-2. ``OPENTELEMETRY_PYTHON_METER_PROVIDER``
+For example, these environment variables will be read:
+
+1. ``OPENTELEMETRY_PYTHON_SOMETHING``
+2. ``OPENTELEMETRY_PYTHON_SOMETHING_ELSE_``
+3. ``OPENTELEMETRY_PYTHON_SOMETHING_ELSE_AND__ELSE``
+
+These won't:
+
+1. ``OPENTELEMETRY_PYTH_SOMETHING``
+2. ``OPENTELEMETRY_PYTHON_something``
+3. ``OPENTELEMETRY_PYTHON_SOMETHING_2_AND__ELSE``
```

Comment:
Is there any specific reason to not support this one?

Reply:
Fixed, any valid variable name is supported now.